doc/compressed_transactions.md

```diff
@@ -0,0 +1,49 @@
+# Compressed Transaction Schema
+
+## 1. Abstraction
+With this transaction compression schema we reduce the size of all transactions by at least 15-20% and up to 40% with signature recovery. The main current uses for this will be steganography or other low bandwidth channels for passing transaction data.
+
+## 2. Methods
+The four main methods to achieve a lower transactions size are:
+1. Adding a few bytes of data before the transaction and each of its inputs and outputs to determine the structure of the following data.
+2. Removing all excess white space and fillers with the use of Variable Integers.
+3. The use of compressed signatures and public key recovery upon decompression
+4. Replacing the 32 byte Txid with a block height and index for the transaction instead.
```

Comment:
In 61504d1a4d7ea545b1880686e426b1acae7d26f0:

I think we could replace the whole 36-byte txid/vout pair with an index into the block.

In principle we could do an ordinal-like thing to index the vouts across all blocks -- but this would require extra database indexes and also might invite fighting/bikeshedding about ordinals.

Reply:
I would suggest using a 24-bit index for the block (up to 16MM blocks, in like 100 years), and a 16-bit index for the vout (up to 64k vouts in a block, which is extremely hard to even fit into a 1MB block). Or varints. Unsure what makes most sense.